src/main/java/fi/otavanopisto/kuntaapi/server/integrations/ptv/translation/PtvOutPtvInTranslator.java

```diff
@@ -328,6 +378,37 @@ private String translateUuid(UUID uuid) {
     return result;
   }
 
+  private V7VmOpenApiAddressDeliveryIn translateDeliveryAddresses(V7VmOpenApiAddressDelivery address) {
+    if (address == null) {
+      return null;
+    }
+    
+    V7VmOpenApiAddressDeliveryIn result = new V7VmOpenApiAddressDeliveryIn();
+    result.setDeliveryAddressInText(address.getDeliveryAddressInText());
+    result.setSubType(address.getSubType());
+    result.setPostOfficeBoxAddress(null);
+    result.setStreetAddress(null);
+    
+    PtvAddressSubtype subtype = getAddressSubtype(result.getSubType());
+    switch (subtype) {
+      case POST_OFFICE_BOX:
+        result.setPostOfficeBoxAddress(translateAddressPostOfficeBox(address.getPostOfficeBoxAddress()));
+      break;
+      case NO_ADDRESS:
+      break;
+      case SINGLE:
+      case STREET:
+        result.setStreetAddress(translateAddressStreet(address.getStreetAddress()));
+      break;
+    
+      default:
+        logger.log(Level.SEVERE, () -> String.format("Unknown subtype %s", result.getSubType()));
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Unknown subtype %s" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)